specs/_features/eip7547/beacon-chain.md

```diff
@@ -0,0 +1,294 @@
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+  - [Domain types](#domain-types)
+- [Preset](#preset)
+  - [Execution](#execution)
+- [Containers](#containers)
+  - [New Containers](#new-containers)
+    - [`SignedInclusionListSummary`](#signedinclusionlistsummary)
+  - [Execution engine](#execution-engine)
+    - [Request data](#request-data)
+      - [New `NewInclusionListRequest`](#new-newinclusionlistrequest)
+    - [Engine APIs](#engine-apis)
+    - [New `notify_new_inclusion_list`](#new-notify_new_inclusion_list)
+  - [Extended containers](#extended-containers)
+    - [`ExecutionPayload`](#executionpayload)
+    - [`ExecutionPayloadHeader`](#executionpayloadheader)
+    - [`BeaconState`](#beaconstate)
+  - [Block processing](#block-processing)
+    - [Block header](#block-header)
+    - [Execution payload](#execution-payload)
+      - [Modified `process_execution_payload`](#modified-process_execution_payload)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to add an inclusion list mechanism to allow forced transaction inclusion. Refers to [EIP-7547](https://eips.ethereum.org/EIPS/eip-7547).
+
+*Note:* This specification is built upon [Deneb](../../deneb/beacon_chain.md) and is under active development.
+
+## Constants
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_INCLUSION_LIST_SUMMARY`     | `DomainType('0x0B000000')` |
+
+## Preset
+
+### Execution
+
+| Name | Value |
+| - | - |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(256)` |
+
+## Containers
+
+### New Containers
+
+#### `SignedInclusionListSummary`
+
+```python
+class SignedInclusionListSummary(Container):
+    summary: List[ExecutionAddress, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+    signature: BLSSignature 
+```
+
+### Execution engine
+
+#### Request data
+
+##### New `NewInclusionListRequest`
+
+```python
+@dataclass
+class NewInclusionListRequest(object):
+    inclusion_list: List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+    summary: List[ExecutionAddress, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+    parent_block_hash: Hash32
+```
+
+#### Engine APIs
+
+#### New `notify_new_inclusion_list`
+
+```python
+def notify_new_inclusion_list(self: ExecutionEngine,
+                              inclusion_list_request: NewInclusionListRequest) -> bool:
+    """
+    Return ``True`` if and only if the transactions in the inclusion list can be successfully executed
+    starting from the execution state corresponding to the `parent_block_hash` in the inclusion list 
+    summary. The execution engine also checks that the total gas limit is less or equal that
+    ``MAX_GAS_PER_INCLUSION_LIST``, and the transactions in the list of transactions correspond to
```

Comment:
Perhaps remove the mention of `MAX_GAS_PER_INCLUSION_LIST` and have this an EL configuration constant mentioned in the EL specification.

Reply:
fair! removing ya, will have it as a EL param